front/pages/api/w/[wId]/assistant/agent_configurations/index.ts

```diff
@@ -116,26 +118,25 @@ async function handler(
         sort,
       });
       if (withUsage === "true") {
-        agentConfigurations = await safeRedisClient(async (redis) => {
-          return Promise.all(
-            agentConfigurations.map(
-              async (
-                agentConfiguration
-              ): Promise<LightAgentConfigurationType> => {
-                return {
-                  ...agentConfiguration,
-                  usage: await getAgentUsage(auth, {
-                    providedRedis: redis,
-                    agentConfiguration,
-                    workspaceId: owner.sId,
-                  }),
-                };
-              }
-            )
-          );
+        const mentionCounts = await safeRedisClient(async (redis) => {
+          return getAgentsUsage({
+            providedRedis: redis,
+            workspaceId: owner.sId,
+          });
         });
+        const usageMap = new Map(
+          mentionCounts.map(({ agentId, count }) => [agentId, count])
+        );
+        agentConfigurations = agentConfigurations.map((agentConfiguration) => ({
+          ...agentConfiguration,
+          usage: {
+            userCount: 0,
+            messageCount: usageMap.get(agentConfiguration.sId) || 0,
+            usersWithAgentInListCount: 0,
+            timePeriodSec: 0,
```

Comment:
Temporary hack. Will probably need to introduce a new type. We don't need `userCount` , `usersWithAgentInListCount` in the main page.

Reply:
@philipperolet what do you think of introducing a new `LightAgentUsageType` and have `LightAgentConfigurationType` infer the type? 
Or simply making `userCount` and `usersWithAgentInListCount` optional in `AgentUsageType`.

Reply:
We still want usage, but just messageCount

Reply:
We basically want to set 'usage.messageCount' but noth the other usage fields. On the homepage we only sort (and display) by 'usage.messageCount'.
 That's why I was proposing to make other fields optional

Doest that make more sense?

Reply:
Just to recap. Right now we have `usage?: AgentUsageType;` with:
```
export type AgentUsageType = {
  userCount: number;
  messageCount: number;
  usersWithAgentInListCount: number;

  // userCount and messageCount are over the last `timePeriodSec` seconds
  timePeriodSec: number;
};
```

You question is how we want to handle the fact that at times we only want to return `messageCount`

What's the future of `userCount` and `usersWithAgentInListCount` ? Do we still support them?

Reply:
> You question is how we want to handle the fact that at times we only want to return messageCount

Exactly 😃 

> What's the future of userCount and usersWithAgentInListCount ? Do we still support them?

Yes we still support them.

Reply:
I would maybe create two types and type the field as a  one or another and adjust methods and endpoints signatures.

But it you run into complexities with this approach I would bot be shocked it we just made the fields optionals (but that adds code client side to check they are defined etc...)

Reply:
IMHO:
- do a quick fix for this PR, simplest is best, e.g. return -1 as count when we don't have it, whatever
- follow-up PR to always compute userCount similarly to message count -- it can be done with the same query since userId is in mentions
- follow-up PR to kill usersInListCount. While we could think it would be a good metric when we added it initially, experience proved it was not relevant, especially since all company assistants are in list by default, and people don't use the list a lot

Reply:
@philipperolet  The current PR already computes `userCount` similarly to `messageCount`.  😃

Reply:
Excellent! In that case I would just kill usersInListCount, and thus avoid the LightAgentUsageType complication :+1:

Reply:
another option if e.g. @Duncid wants to keep it would be to compute the usersInListCount too, it's just another SQL query every 2 hours (1 query for all agents) so it's not actually increasing load

Reply:
@philipperolet The main reason I actually introduced this new `LightAgentUsageType` is to speed up retrieval when landing on the homepage: we only retrieve the `messageCount` in Redis (as this is the only usage metric used there) and don't need to retrieve `userCount` from Redis nor `usersWithAgentInListCount` in DB.

Reply:
As per IRL, retrieving userCount is not costful;  as for usersWithAgentInListCount we can either kill it if ed is ok or compute it cheaply if not (happy to help with the 'cheaply' part here)